cli/context/store/tlsstore.go

```diff
@@ -0,0 +1,84 @@
+package store
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+const tlsDir = "tls"
+
+type tlsStore struct {
+	root string
+}
+
+func (s *tlsStore) contextDir(name string) string {
+	return filepath.Join(s.root, name)
+}
+
+func (s *tlsStore) endpointDir(contextName, name string) string {
+	return filepath.Join(s.root, contextName, name)
+}
+
+func (s *tlsStore) filePath(contextName, endpointName, filename string) string {
+	return filepath.Join(s.root, contextName, endpointName, filename)
+}
+
+func (s *tlsStore) createOrUpdate(contextName, endpointName, filename string, data []byte) error {
+	epdir := s.endpointDir(contextName, endpointName)
+	if err := os.MkdirAll(epdir, 0700); err != nil {
```

Comment:
We can't do an MKDir there, because endpointName is a subdir of $context_root/tls/$context_name.
However, tlsStore is not exposed and can only be created trough store.New, which creates the directory structure with correct permissions.

Reply:
see https://github.com/docker/cli/pull/1500/files#diff-52f67b7303ba5e1a6c9cd3ec76a84e0cR79